common/navigator/core/aws.py

```diff
@@ -150,6 +151,67 @@ def move_document(
             new_bucket, os.getenv("AWS_REGION"), new_key or s3_document.key
         )
 
+    def list_files(self, bucket: str, max_keys=1000) -> t.Union[S3Document, bool]:
+        """Yields the documents contained in a bucket on S3
+
+        Args:
+            bucket (str): name of the bucket in which the files will be listed.
+            max_keys (int): maximum number of files to return in a single call
+
+        Returns:
+            False if the operation was unsuccessful.
+
+        Yields:
+            S3Document: representing each document.
+        """
+
+        is_truncated = True
+        next_continuation_token = None
+        try:
+            while is_truncated:
+                # Include a continuation token in the arguments to boto3 list_objects_v2
+                # if we want to continue listing files from the last call
+
+                kwargs = {"Bucket": bucket, "MaxKeys": max_keys}
```

Comment:
My interpretation of the `max_keys` kwarg from the docstring was that it would limit the number of returned documents: I interpreted "maximum number of files to return in *a single call*" as being a single call of the `list_files` method. 

From looking at the code, it seems like it essentially controls the page size of the `list_files` method. 

Could you please update the docstring to clarify this? 

More generally, I think using 'a single call' to refer to a call of a function within the method being documented is probably bad practice, as it requires understanding the implementation of the function.

Reply:
Change made